stories/webapp/hierarchy/ItemRow/NewHierarchyItemRow.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import { func, string } from 'prop-types';
+import { Icon, ItemRow } from 'lib';
+import { HierarchyRowForm } from '../HierarchyRowForm';
+import { HierarchyItemRow } from './HierarchyItemRow';
+
+const NewHierarchyItemRow = ({ id, removeItem }) => {
+  const [itemName, setItemName] = useState(null);
+  const [isEditing, setIsEditing] = useState(true);
+  const [isSaving, setIsSaving] = useState(false);
+  const [hasSaved, setHasSaved] = useState(false);
```

Comment:
Im sure most of this would be replaced when it comes time to hook up to an api, but maybe a status state might be a bit cleaner? so:
`const [currentStatus, setStatus] = useState('');` then the status can be set to `IS_SAVING`, `IS_EDITING`, `HAS_SAVED`. Since it doesnt look like any two need to be true at any given time. Or perhaps just ` `, `IS_SAVING`, `HAS_SAVED` to get rid of having both isSaving and hasSaved states.

Not something you have to action, just felt a bit weird to me!